libbeat/processors/add_cloud_metadata/provider_hetzner_cloud_test.go

```diff
@@ -30,8 +31,16 @@ import (
 	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+func init() {
+	// Disable IMDS when the real AWS SDK IMDS client is used,
+	// so tests are isolated from the environment. Otherwise,
+	// tests for non-EC2 providers may fail when the tests are
+	// run within an EC2 VM.
+	os.Setenv("AWS_EC2_METADATA_DISABLED", "true")
+}
```

Comment:
Yep!! Sorry I was testing for hetzner and wasn't thinking adding into `provider_aws_ec2.go` will be the centralized location instead of adding it per test... Thanks for the review!

Reply:
Sorry I miswrote, I meant provider_aws_ec2_test.go! We definitely don't want to disable it for production :)